python/smqtk/algorithms/saliency/rise.py

```diff
@@ -91,21 +94,30 @@ def generate(self, base_image, augmenter, descriptor_generator,
             salient regions according to the given blackbox algorithm.
         :rtype: numpy.ndarray[float]
         """
-        # augmenter = RISEAugmenter(1000, 10, 0.5, (224, 224))
-        masked_images, masks = augmenter.augment(base_image)
         
-        scores = np.zeros(masked_images.shape[0])
-        # Performance-critical part since we run the model on a large number of images.
-        # Should be well-optimized, and should be done in batches if possible.
-        for i, masked_image in enumerate(masked_images):
-            desc = descriptor_generator.compute_descriptor(masked_image)
-            score = blackbox.transform(desc)
-            scores[i] = score
+        resized_img = resize(image_mat, augmenter.input_size, order=1)
+        masked_images, masks = augmenter.augment(resized_img)
+        
+        idx_to_uuid = []
+        def iter_aug_img_data_elements():
+            for a in masked_images:
+                buff = six.BytesIO()
+                (a).save(buff, format="bmp")
+                de = DataMemoryElement(buff.getvalue(),
+                                       content_type='image/bmp')
+                idx_to_uuid.append(de.uuid())
+                yield de
+
+        uuid_to_desc = descriptor_generator.compute_descriptor_async(iter_aug_img_data_elements())
+        scores = blackbox.transform((uuid_to_desc[uuid] for uuid in idx_to_uuid))
+        
         # Compute a weighted average of masks w.r.t. the scores
         saliency_map = np.average(masks, axis=0, weights=scores)
         saliency_map = np.squeeze(saliency_map)
         # Normalize
         saliency_map /= masks.mean(axis=0)
+        # Resize back to the original image shape
+        saleincy_map = resize(saliency_map, image_mat.shape, order=1)
```

Comment:
'saleincy_map', looks like a typo. Once corrected, this can be merged into dev/XAI, Thank you for working through the requested changes.